hearth/spec/hearth/connection_pool_spec.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+module Hearth
+  describe ConnectionPool do
+    # Set instance variable instead of calling empty!
+    # to avoid re-use of double rspec error.
+    before do
+      ConnectionPool.instance_variable_set(:@pools, {})
+    end
+
+    let(:config) { { timeout: 1 } }
+    let(:pool) { ConnectionPool.for(config) }
+
+    let(:endpoint) { URI('https://example.com') }
+    let(:endpoint2) { URI('https://example.org') }
+    let(:endpoint_path_query) { URI('https://example.com/path?query') }
+    let(:connection) { double('connection', stale?: false, finish: nil) }
+    let(:connection2) { double('connection2', stale?: false, finish: nil) }
+    let(:stale_connection) { double('stale_connection', stale?: true) }
+
+    shared_examples 'thread safety' do
+      it 'is thread safe' do
```

Comment:
It did once in many runs, but you're right, I don't think this is correct. I tried expecting Mutex too, but there's funkiness because there's two of them and they get nested. I think it's better to leave these tests out.